src/plugins/dashboard/public/dashboard_container/embeddable/dashboard_container.tsx

```diff
@@ -176,6 +177,10 @@ export class DashboardContainer
     this.dashboardCreationStartTime = dashboardCreationStartTime;
 
     // start diffing dashboard state
+    this.unsavedChanges = new BehaviorSubject(false);
+    this.backupUnsavedChanges = new BehaviorSubject<Partial<DashboardContainerInput> | undefined>(
+      undefined
+    );
```

Comment:
Originally, I tried to have only **one** behaviour subject + subscription for unsaved changes - however, this got complicated with session storage backup because the keys that count as unsaved changes are **not** the same keys that should be backed up. It made the logic of the subscription... very messy, so separating it into two different behaviour subjects + subscriptions felt **a lot** cleaner. Open to other suggestions, though :+1: